### DISCUSSION TITLE
Persistent data with Jotai

### DISCUSSION THREAD

**Discussion by ghost** [pmndrs/jotai#350] [Q&A] [answered]
I'm writing a toy react app, found using jotai to hold values accessible from a different components pretty intuitive even I'm newbie.

I  need changed values to persist during page reload and all subsequent visits, I think indexdb/localstorage is the way to achieve this,  so how can I get values from indexdb/localstorage everytime and store in atoms when the app launched?

**Comment by Aslemammad** [comment, marked as answer]
Hello @grubeli, You can check [this](https://github.com/pmndrs/jotai/blob/master/docs/guides/persistence.md).
```ts
const strAtom = atom(localStorage.getItem('myKey') ?? 'foo')

const strAtomWithPersistence = atom(
  (get) => get(strAtom),
  (get, set, newStr) => {
    set(strAtom, newStr)
    localStorage.setItem('myKey', newStr)
  }
)
```

**Reply by ghost** [reply]
Amazing.
Maybe offtopic, is there a similar way to use with IndexDB since localStorage have 5/10MB storage limit?

**Reply by Aslemammad** [reply]
It's the same approach; I just showed you the idea.
